addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java

```diff
@@ -0,0 +1,1061 @@
+            if (command.equals(OnOffType.OFF)) {
+                // FIXME Need command format
+                sendCommand(macAddress, ";LIGHT;SMART;OFF");
+            } else if (command.equals(OnOffType.ON)) {
+                sendCommand(macAddress, ";LIGHT;SMART;ON");
+            }
+        }
+    }
+
+    private void handleLightLevelMin(Command command) {
+        logger.debug("Handling light level minimum command {}", command.toString());
+        // <mac;LIGHT;LEVEL;MIN;0-16>
+        if (command instanceof PercentType) {
+            // Send min light level set command
+            sendCommand(macAddress, ";LIGHT;LEVEL;MIN;".concat(convertPercentToLevel(command)));
+            // Don't let max be less than min
+            adjustMaxLevel(command);
+        }
+    }
+
+    private void handleLightLevelMax(Command command) {
+        logger.debug("Handling light level maximum command {}", command.toString());
+        // <mac;LIGHT;LEVEL;MAX;0-16>
+        if (command instanceof PercentType) {
+            // Send max light level set command
+            sendCommand(macAddress, ";LIGHT;LEVEL;MAX;".concat(convertPercentToLevel(command)));
+            // Don't let min be greater than max
+            adjustMinLevel(command);
+        }
+    }
+
+    private void adjustMaxLevel(Command command) {
+        int newMin = ((PercentType) command).intValue();
+        int currentMax = PercentType.ZERO.intValue();
+        if (fanStateMap.get(CHANNEL_LIGHT_LEVEL_MAX) != null) {
+            currentMax = ((PercentType) fanStateMap.get(CHANNEL_LIGHT_LEVEL_MAX)).intValue();
+        }
+        if (newMin > currentMax) {
+            updateState(CHANNEL_LIGHT_LEVEL_MAX, (PercentType) command);
+            sendCommand(macAddress, ";LIGHT;LEVEL;MAX;".concat(convertPercentToLevel(command)));
+        }
+    }
+
+    private void adjustMinLevel(Command command) {
+        int newMax = ((PercentType) command).intValue();
+        int currentMin = PercentType.HUNDRED.intValue();
+        if (fanStateMap.get(CHANNEL_LIGHT_LEVEL_MIN) != null) {
+            currentMin = ((PercentType) fanStateMap.get(CHANNEL_LIGHT_LEVEL_MIN)).intValue();
+        }
+        if (newMax < currentMin) {
+            updateState(CHANNEL_LIGHT_LEVEL_MIN, (PercentType) command);
+            sendCommand(macAddress, ";LIGHT;LEVEL;MIN;".concat(convertPercentToLevel(command)));
+        }
+    }
+
+    /*
+     * Convert from fan range (0-7) and light range (0-16) to dimmer range (0-100).
+     */
+    private String convertPercentToSpeed(Command command) {
+        // Produces values 0-7 from dimmer state
+        return String.valueOf((int) Math.round(((PercentType) command).doubleValue() / 14.2857));
+    }
+
+    private PercentType convertSpeedToPercent(String speed) {
+        // Produces dimmer state from values 0-7
+        return new PercentType((int) Math.round(Integer.parseInt(speed) * 14.2857));
+    }
+
+    private String convertPercentToLevel(Command command) {
```

Comment:
Maybe its wise to call this method with a PercentType now the cast is potentially unchecked